### PR TITLE
Replace H256 with PoolId/DelegationId

### PR DESCRIPTION
**Pull request by azarovh**

### CODE REVIEW

File: pos_accounting/src/lib.rs [hunk 1]
```diff
@@ -1,3 +1,24 @@
+use common::primitives::H256;
+use serialization::{Decode, Encode};
+
 pub mod error;
 pub mod pool;
 pub mod storage;
+
+#[derive(Clone, Copy, Debug, Encode, Decode, Default, Eq, PartialEq, Ord, PartialOrd)]
+pub struct PoolId(H256);
```
Comment (azarovh) [inline pos_accounting/src/lib.rs:9]: not sure whether I should move these new types to `id.rs` or leave here

Comment (TheQuantumPhysicist) [reply pos_accounting/src/lib.rs:9]: Perhaps the better way to do this is defining an empty struct, like Pool, and doing `Id<Pool>`. Then we can later decide what we want to attach this to. 

The problem with the current format is that you have to redefine everything we defined for `Id<>`

**Review by TheQuantumPhysicist — APPROVED** [review]
Looking good. Just please look into PoolId and DelegationId definitions as I mentioned in the comment

**Comment by azarovh** [comment]
> Just please look into PoolId and DelegationId definitions as I mentioned in the comment

Confused what do you mean? I defined empty structs as you suggested

**Review by stanislav-tkach — APPROVED** [review]

**Comment by TheQuantumPhysicist** [comment]
> > Just please look into PoolId and DelegationId definitions as I mentioned in the comment
> 
> Confused what do you mean? I defined empty structs as you suggested

All good. I submitted the comment late.

**Pull request merged**